python_modules/libraries/dagster-components/dagster_components/resolved/model.py

```diff
@@ -118,11 +125,69 @@ class AttrWithContextFn:
     callable: Callable[["ResolutionContext", Any], Any]
 
 
+@dataclass(frozen=True)
+class ResolveFromModel:
+    """Annotated marker to specify that the target type should be resolved from its model.
+
+    The via argument allows use for target types that are not directly ResolvableFrom.
+    """
+
+    via: Optional[type[ResolvedKwargs]] = None
+
+
+class ResolveFromInjection:
```

Comment:
I anticipate that we will want to rename this (and Injected) in the near future. As it stands, this is quite vague. I think the name should potentially include a reference to the fact that there's a string involved in the process somewhere?